wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -764,16 +764,16 @@ def icon(name=None, classname=None, title=None, wrapped=False, class_name=None):
 
         warn(
             (
-                "Icon template tag `class_name` has been renamed to `classname`, please adopt the new usage instead.",
-                f'Replace `{{% icon ... class_name="{class_name}" %}}` with `{{% icon ... classname="{class_name}" %}}`',
+                "Icon template tag `class_name` has been renamed to `classname`, please adopt the new usage instead. "
+                f'Replace `{{% icon ... class_name="{class_name}" %}}` with `{{% icon ... classname="{class_name}" %}}`'
             ),
             category=RemovedInWagtail50Warning,
         )
 
     return {
         "name": name,
         # supporting class_name for backwards compatibility
-        "classname": classname or class_name or "icon",
+        "classname": classname or class_name or "",
```

Comment:
We already have a hardcoded `icon` class in the HTML:

https://github.com/wagtail/wagtail/blob/357edf2914bfb7b1265011b1009f0d57ff9824f1/wagtail/admin/templates/wagtailadmin/shared/icon.html#L4

Adding `or "icon"` would add an extraneous `icon` class to the rendered HTML.